### PR TITLE
International Stores: add buy box when appropriate

### PR DESCRIPTION
**Pull request by danielbeardsley**
This should function the same as it did on the previous product page.

* Makes an API request to ifixit.com from the browser for the given
  product
* That responds with "user is geolocated in country X, here's info for a
  banner for each of the variants"
  * OR it responds with null
* We use a react component to render the "Buy from our X Store" banner
  for the currently selected variant
* Shows a "or buy from the US Store" link that will dismiss the buy box
  for 7 days
   * The previous product page used a 7-day cookie, so I kept it the
     same.

Note: The UI differs from the old product page bannerl slightly and we
can definitely tweak it, but I might do that in a follow-up pull unless
there are bugs.

Also, matomo events will be in a follow up pull.

QA:
* Set your preferred store via the store dropdown in the header
* Visit product pages on the US store for products that are available in your preferred store
* Assert you see the "buy box" instead of the Add To Cart button
* Assert that clicking the button takes you to the correct store
* Assert that clicking the bottom option "buy from US store" makes the buy box disappear and you shouldn't see it on further page reloads (should stay away for 7 days)

Connect #930

### CODE REVIEW

File: frontend/lib/ifixit-api/international-buy-box.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+   salesChannel: string;
+   // Keyed by optionid
+   products: Record<string, BuyBoxVariant>;
+};
+
+export function getBuyBoxForProduct(
+   client: IFixitAPIClient,
+   productcode: string
+): Promise<BuyBoxResponse | null> {
+   // TODO pull country override from browser url
```
Comment (sterlinghirsh) [inline frontend/lib/ifixit-api/international-buy-box.ts:29]: Should this be done? Not sure how we'll QA without this.

**Review by sterlinghirsh — DISMISSED** [review]
CR :zap: though this is missing the country override code. There are also no QA instructions.

File: frontend/lib/ifixit-api/international-buy-box.ts (hunk 1 above)
Comment (danielbeardsley) [reply frontend/lib/ifixit-api/international-buy-box.ts:29]: I'm pretty sure setting `country=` is just a shortcut to changing your preferred store. When we "geo-locate" you, we use the preferred store if it's set.

**Comment by danielbeardsley** [comment]
QA instructions added

**Comment by danielbeardsley** [comment]
Here's a link to the preferred store logic for reference:
https://github.com/iFixit/ifixit/blob/e7407b612e1efdd31d57645971608568c2938688/iFixit/Cart/Stores/StoreProvider.php#L304-L352

**Comment by deltuh-vee** [comment]
If I clear my cookies on the react commerce product page that previously had a buy box, should I see the buy box appear again? This is after clicking "Buy from US Store"

**Comment by deltuh-vee** [comment]
dev_block ✊ 
> When the International Buy Box is present / unresolved, we must hide the "Frequently Bought Together" section to ensure no "Add to Cart" action is available on the page.

This section is still available with the International Buy Box present.
image

**Comment by sterlinghirsh** [comment]
Can we add the logic to specify country= in the url?

**Comment by danielbeardsley** [comment]
> Can we add the logic to specify country= in the url?

That parameter doesn't actually work. It likely used to.

https://www.cominor.com/api/2.0/internal/international_store_promotion/buybox?country=CA&productcode=145198
https://www.cominor.com/api/2.0/internal/international_store_promotion/buybox?country=DE&productcode=145198

In practice, it's not much harder to just change your preferred store.

**Comment by danielbeardsley** [comment]
un_dev_block :+1:    Thanks for catching that @deltuh-vee !  Fixed!

**Comment by sterlinghirsh** [comment]
CR :zap: dev_block :zap: on https://react-commerce-git-intl-buy-box-ifixit.cominor.com/products/iopener we correctly hide "frequently bought together" but after clicking "buy from US store", it doesn't reappear. After refreshing the page, it reappears.

**Comment by danielbeardsley** [comment]
un_dev_block :+1: 

Good catch @sterlinghirsh 

I wanted so much to use redux or make a little state store implementation (like https://github.com/iFixit/ifixit/blob/master/carpenter-frontend/Shared/use-shared-state.ts), but I resisted and just went with calling it higher up in the stack and passing props down.

**Review by sterlinghirsh — APPROVED** [review]
CR :zap:

**Comment by sterlinghirsh** [comment]
QA ⚡

**Pull request merged**